README.md
# VKBK
Инструмент для создания и синхронизации локального бэкапа вашего лампового профиля ВКонтакте.

### Актуальная версия 0.9.1

### Системные требования
Софт | Версия | Модули
--- | --- | ---
Apache | 2.4+ | 
PHP | 7.3+ | curl, mbstring, mysqli, json
MySQL | 5.5+ | 
Composer | 1.10.6+ | twig, vk-auth, guzzle

## Инсталляция
[Инстукция по установке и настройке](https://github.com/Chiaki/VKBK/wiki/Установка-и-настройка)

## История версий
Посмотреть [историю версий](CHANGELOG.md) проекта VKBK.

## Функционал
На данный момент реализован следующий функционал:
+ Получение диалогов (личные сообщения) (тип: фото, видео, ссылки, документы, стикеры, репосты стен и сообществ, цитирование)
- Получение альбомов (в том числе и системных)
- Получение фотографий
- Получение документов
- Получение аудиозаписей ~~и альбомов~~
> ~~В версии 0.8.5 снова появилась возможность сохранять свои аудиозаписи.~~ ВК опять всё поломали. ):
- Получение видеозаписей
> (просмотр через плеер, возможность создать локальную копию при помощи [youtube-dl](https://github.com/rg3/youtube-dl))
* Получение сообщений, комментариев и репостов со стены (тип: фото, видео, ссылки, документы, стикеры, ~~музыка~~)

Через веб-интерфейс возможен просмотр диалогов, альбомов, фотографий, стены, прослушивание аудиозаписей, просмотр видеозаписей через локальный и сторонние плееры.



CHANGELOG.md
## v0.9.1
-   [upd] В синхронизацию видео добавлен дополнительный этап для получения приватных видео.

## v0.9.0
-   [new] Добавлена синхронизация комментариев

## v0.8.13
-   [new] Добавлен исталлятор
-   [new] Стена: добавлена возможность поиска текста и фильтр по дате.
-   [new] Добавлен Air Datepicker
-   [upd] Небольшие правки
-   [fix] Исправлена ошибка при сохранении превью для видео в некоторых случаях.
-   [fix] Исправлена ошибка при сохранении видео со стены.

## v0.8.12
-   [info] Проект потихоньку переползает на использование composer. (: В качестве шаблонизатора был вырбран twig.
-   [new] В качестве альтернативного варианта доступа к API добавлен vk-auth.
-   [new] Синхронизация диалогов снова работает.
-   [upd] Исправлено сохранение стикеров.

## v0.8.11
-   [new] Обновлена синхронизация видео под API 5.103.
-   [upd] Обновлена синхронизация фото альбомов под API 5.103.
-   [upd] Правки для youtube-dl.

## v0.8.10
-   [new] Синхронизация стены теперь на ajax.

## v0.8.9
-   [new] Обновлен раздел синхронизации. Теперь он работает на ajax.
-   [upd] Bootstrap обновлен до версии 4.3.1
-   [upd] Обновление popper.js для Bootstrap от CloudFlare

## v0.8.8
-   [upd] fancybox.js обновлен до версии 3.5.7
-   [upd] Font Awesome обновлен до версии 5.8.2
-   [upd] jQuery обновлен до версии 3.4.1
-   [upd] perfect-scrollbar.js обновлен до версии 1.4.0
-   [upd] popper.js обновлен до версии 1.15.0
-   [upd] Plyr обновлен до версии 3.5.4

## v0.8.7b
-   [fix] Исправлена ошибка в проверке типа для очереди закачки.
-   [fix] Исправлено отображение документа в сообщениях (стене).

## v0.8.7a
-   [info] Добавлена проверка необходимых модулей для работы скрипта.
-   [info] Используется VK API версии 5.92.
-   [new] В синхронизации диалогов добавлено сохранение цитируемых\пересылаемых постов со стены.
-   [upd] Добавлена возможность исключать из очереди фотографии.
-   [upd] Старая версия API при синхронизации диалогов не всегда отдавала самое лучшее качество для изображений.
-   [upd] FontAwesome обновлен до версии 5.6.3
-   [fix] Некорректное добавление новых групп при синхронизации диалогов.
-   [fix] Потеря информации о локальной копии файла в диалогах.
-   [upd] Мелкие изменения и исправления.

## v0.8.6
-   [new] В синхронизации диалогов добавлено сохранение цитируемых\пересылаемых сообщений.
-   [upd] В диалогах добавлена группировка сообщений от одного пользователя.
-   [upd] Мелкие изменения и исправления.

## v0.8.5
-   [new] Возможность сохранения музыки возвращается. К сожалению не в полноценном виде как API, и неизвестно на какое время, но пока работает. Спасибо за это @in4in-dev ♥
-   [new] Плеер локального видео jPlayer заменен на Plyr.
-   [fix] В разделе видео при наличии в url хэштег параметров, страницы не подгружались.
-   [fix] Групповые диалоги не работали корректно.
-   [fix] Потеря информация о локальной копии файла в диалогах.
-   [upd] Мелкие изменения и исправления.

## v0.8.4
-   [fix] Обновлена кодировка некоторых таблиц для поддержки эмодзи.
-   [upd] Bootstrap обновлен до версии 4.1.3
-   [upd] Bootstrap Select обновлен до версии 1.13.2
-   [upd] FontAwesome обновлен до версии 5.3.1
-   [upd] js-cookie обновлен до версии 2.2.0
-   [upd] Мелкие изменения.

## v0.8.3
-   [fix] Исправлено отображение незнакомых пользователей в групповых диалогах.
-   [upd] Включен просмотр групповых диалогов.

## v0.8.2
-   [new] Добавлена возможность скрывать элементы в очереди закачки (пока для вложений)
-   [fix] Исправлена ошибка при синхронизации видео, когда у видеозаписей совпадал ID.
-   [fix] Исправлены мелкие баги

## v0.8.1
-   [fix] Обновлено регулярное выражение для профилей и групп в связи с изменением url ВК.

## v0.8.0
-   [new] Синхронизация личных сообщений (диалогов)
-   [upd] perfect-scrollbar обновлен до версии 1.3.0
-   [upd] Исправлены мелкие недоработки в дизайне
-   [fix] Исправлены мелкие баги

## v0.7.7
-   [upd] Исправлены мелкие недоработки в дизайне
-   [fix] Исправлена ошибка при сохранении файлов в несуществующую директорию. Спасибо @phatal

## v0.7.6
-   [info] Для работы потребуется MySQL версии 5.5+ (с поддержкой utf8mb4)
-   [new] Добавлен Justified Gallery
-   [new] Добавлен popper.js
-   [upd] jQuery обновлен до версии 3.3.1
-   [upd] Bootstrap обновлен до версии 4.0.0
-   [upd] pjax обновлен до версии 2.0.1
-   [upd] jScroll обновлен до версии 2.3.9
-   [upd] Bootstrap Select обновлен до версии 1.13.0-beta

## v0.7.5
-   [upd] Bootstrap-select обновлен до версии 1.12.4
-   [upd] FontAwesome обновлен до версии 5.0.8
-   [info] Очередное обновление API Вконтакта.

## v0.7.4
-   [fix] Исправлена ошибка в разделе Сообщения.

## v0.7.3
-   [fix] Частично исправлен fancybox в разделе сообщений. Для некоторых типов сообщений могут остаться ошибки.
-   [upd] Убраны абсолютные пути из радиректов.

## v0.7.2
-   [new] Видео теперь можно минимизировать и при переходе в другой раздел оно не закроется.
-   [new] Добавлен быстрый поиск для видео.
-   [new] Настройки видео плеера теперь сохраняются в куки.
-   [new] Добавлен плагин js-cookie.
-   [upd] jQuery обновлен до версии 1.12.4
-   [upd] Fancybox обновлен до версии 3.1.20

## v0.7.1
-   [info] При ошибках скачивания видео, обновите youtube-dl до версии от 2017.05.29 или выше.
-   [upd] jScroll обновлен до версии 2.3.9
-   [upd] Bootstrap обновлен до версии 3.3.7
-   [info] Запросы изменены на MySQLi.
-   [fix] Микрофикс ссылок аватаров групп и пользователей в очереди закачки.
-   [bug] Проблема с отображением записи &laquo;пользователь обновил фото&raquo;.

## v0.7.0
-   [new] Добавлено автоматическое проигрывание gif'ов на стене.
-   [new] Добавлена синхронизация документов на стене.
-   [new] Добавлена библиотека pjax.

about.php
<?php

header('Content-Type: text/html; charset=UTF-8');
header("Cache-Control: no-store, no-cache, must-revalidate, max-age=0");
header("Cache-Control: post-check=0, pre-check=0", false);
header("Pragma: no-cache");

require_once('./cfg.php');
$c = new vkbk();

if(isset($_GET['_pjax']) || isset($_POST['_pjax'])){ $cfg['pj'] = true; }

// Get DB
require_once(ROOT.'classes/db.php');
$db = new db();
$res = $db->connect($cfg['host'],$cfg['user'],$cfg['pass'],$cfg['base']);

// Get Skin
require_once(ROOT.'classes/skin.php');
$skin = new skin();

// TWIG
$skin->twig_init($c);

// Get local counters for top menu
$lc = $db->query_row("SELECT * FROM vk_counters");

if(!$cfg['pj']){
	print $skin->header(array('extend'=>''));
	print $skin->navigation($lc);
}

$libs = array(
	array('name' => 'Bootstrap', 			'ver' => '4.1.3', 		'url' => 'https://getbootstrap.com/'),
	array('name' => 'PHP класс для VK.API', 'ver' => '0.1.7', 		'url' => 'https://github.com/vladkens/VK'),
	array('name' => 'vk-auth',              'ver' => '@2c38f41', 	'url' => 'https://github.com/ivastly/vk-auth'),
	array('name' => 'Freewall.js', 			'ver' => '1.05', 		'url' => 'https://github.com/kombai/freewall'),
	array('name' => 'jScroll.js', 			'ver' => '2.3.9a', 		'url' => 'https://github.com/pklauzinski/jscroll'),
	array('name' => 'fancybox.js', 			'ver' => '3.5.7', 		'url' => 'https://github.com/fancyapps/fancyBox'),
	array('name' => 'Font Awesome', 		'ver' => '5.8.2', 		'url' => 'https://fontawesome.com/'),
	array('name' => 'Bootstrap Toggle', 	'ver' => '2.2.2', 		'url' => 'https://github.com/minhur/bootstrap-toggle'),
	array('name' => 'hashnav.js', 			'ver' => '6 May 2016',	'url' => 'https://habr.com/ru/sandbox/57659/'),
	array('name' => 'perfect-scrollbar.js',	'ver' => '1.4.0', 		'url' => 'https://github.com/noraesae/perfect-scrollbar'),
	array('name' => 'jPlayer.js', 			'ver' => '2.9.2', 		'url' => 'https://github.com/happyworm/jPlayer'),
	array('name' => 'Bootstrap Select', 	'ver' => '1.13.2', 		'url' => 'https://github.com/snapappointments/bootstrap-select'),
	array('name' => 'pjax', 				'ver' => '2.0.1', 		'url' => 'https://github.com/defunkt/jquery-pjax'),
	array('name' => 'jQuery Visible', 		'ver' => '1.2.0', 		'url' => 'https://github.com/customd/jquery-visible'),
	array('name' => 'Debounce plugin', 		'ver' => '1.1', 		'url' => 'http://benalman.com/projects/jquery-throttle-debounce-plugin/'),
	array('name' => 'js-cookie', 			'ver' => '2.2.0', 		'url' => 'https://github.com/js-cookie/js-cookie'),
	array('name' => 'popper.js', 			'ver' => '1.15.0', 		'url' => 'https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.15.0/umd/popper.min.js'),
	array('name' => 'Justified Gallery', 	'ver' => '3.7.0', 		'url' => 'https://github.com/miromannino/Justified-Gallery'),
	array('name' => 'Plyr', 				'ver' => '3.5.4', 		'url' => 'https://github.com/sampotts/plyr'),
	array('name' => 'Air Datepicker', 		'ver' => '@3e73721', 	'url' => 'https://github.com/t1m0n/air-datepicker'),
);

$changelog = array(
	'0.9.x' => array(
		'0.9.1' => array(
			array('u',"В синхронизацию видео добавлен дополнительный этап для получения приватных видео."),
		),
		'0.9.0' => array(
			array('n',"Добавлена синхронизация комментариев"),
		),
	),
	'0.8.x' => array(
		'0.8.13' => array(
			array('n',"Добавлен исталлятор"),
			array('n',"Стена: добавлена возможность поиска текста и фильтр по дате."),
			array('n',"Добавлен Air Datepicker"),
			array('u',"Небольшие правки"),
			array('bf',"Исправлена ошибка при сохранении превью для видео в некоторых случаях."),
			array('bf',"Исправлена ошибка при сохранении видео со стены."),
		),
		'0.8.12' => array(
			array('i',"Проект потихоньку переползает на использование composer. (: В качестве шаблонизатора был вырбран twig."),
			array('n',"В качестве альтернативного варианта доступа к API добавлен vk-auth."),
			array('n',"Синхронизация диалогов снова работает."),
			array('u',"Исправлено сохранение стикеров."),
		),
		'0.8.11' => array(
			array('u',"Обновлена синхронизация видео под API 5.103."),
			array('u',"Обновлена синхронизация фото альбомов под API 5.103."),
			array('u',"Правки для youtube-dl."),
		),
		'0.8.10' => array(
			array('n',"Синхронизация стены теперь на ajax."),
		),
		'0.8.9' => array(
			array('n',"Обновлен раздел синхронизации. Теперь он работает на ajax."),
			array('u','Bootstrap обновлен до версии 4.3.1'),
			array('u','Обновление popper.js для Bootstrap от CloudFlare'),
		),
		'0.8.8' => array(
			array('u','fancybox.js обновлен до версии 3.5.7'),
			array('u','Font Awesome обновлен до версии 5.8.2'),
			array('u','jQuery обновлен до версии 3.4.1'),
			array('u','perfect-scrollbar.js обновлен до версии 1.4.0'),
			array('u','popper.js обновлен до версии 1.15.0'),
			array('u','Plyr обновлен до версии 3.5.4'),
		),
		'0.8.7' => array(
			array('i',"Добавлена проверка необходимых модулей для работы скрипта."),
			array('i',"Используется VK API версии 5.92."),
			array('n',"В синхронизации диалогов добавлено сохранение цитируемых\пересылаемых постов со стены."),
			array('u',"Добавлена возможность исключать из очереди фотографии, документы."),
			array('u',"Старая версия API при синхронизации диалогов не всегда отдавала самое лучшее качество для изображений."),
			array('u','FontAwesome обновлен до версии 5.6.3'),
			array('bf',"Некорректное добавление новых групп при синхронизации диалогов."),
			array('bf',"Потеря информации о локальной копии файла в диалогах."),
			array('bf',"Исправлена ошибка в проверке типа для очереди закачки."),
			array('bf',"Исправлено отображение документа в сообщениях (стене)."),
			array('u',"Мелкие изменения и исправления."),
		),
		'0.8.6' => array(
			array('n',"В синхронизации диалогов добавлено сохранение цитируемых\пересылаемых сообщений."),
			array('u',"В диалогах добавлена группировка сообщений от одного пользователя."),
			array('u',"Мелкие изменения и исправления."),
		),
		'0.8.5' => array(
			array('n',"Возможность сохранения музыки возвращается. К сожалению не в полноценном виде как API, и неизвестно на какое время, но пока работает. Спасибо за это @in4in-dev ♥"),
			array('n',"Плеер локального видео jPlayer заменен на Plyr."),
			array('bf',"В разделе видео при наличии в url хэштег параметров, страницы не подгружались."),
			array('bf',"Групповые диалоги не работали корректно."),
			array('bf',"Потеря информация о локальной копии файла в диалогах."),
			array('u',"Мелкие изменения и исправления."),
		),
		'0.8.4' => array(
			array('bf',"Обновлена кодировка некоторых таблиц для поддержки эмодзи."),
			array('u','Bootstrap обновлен до версии 4.1.3'),
			array('u','Bootstrap Select обновлен до версии 1.13.2'),
			array('u','FontAwesome обновлен до версии 5.3.1'),
			array('u','js-cookie обновлен до версии 2.2.0'),
			array('u',"Мелкие изменения."),
		),
		'0.8.3' => array(
			array('bf',"Исправлено отображение незнакомых пользователей в групповых диалогах."),
			array('u',"Включен просмотр групповых диалогов."),
		),
		'0.8.2' => array(
			array('n',"Добавлена возможность скрывать элементы в очереди закачки (пока для вложений)"),
			array('bf',"Исправлена ошибка при синхронизации видео, когда у видеозаписей совпадал ID"),
			array('bf',"Исправлены мелкие баги"),
		),
		'0.8.1' => array(
			array('bf',"Обновлено регулярное выражение для профилей и групп в связи с изменением url ВК."),
			array('bf',"Исправлены мелкие баги"),
		),
		'0.8.0' => array(
			array('n','Синхронизация личных сообщений (диалогов)'),
			array('u','perfect-scrollbar обновлен до версии 1.3.0'),
			array('u',"Исправлены мелкие недоработки в дизайне"),
			array('bf',"Исправлен баг с включением быстрой синхронизации для фотографий."),
			array('bf',"Исправлены мелкие баги"),
		),
	),
	'0.7.x' => array(
		'0.7.7' => array(
			array('u',"Исправлены мелкие недоработки в дизайне"),
			array('bf',"Исправлена ошибка при сохранении файлов в несуществующую директорию. Спасибо @phatal"),
		),
		'0.7.6' => array(
			array('i',"Для работы потребуется MySQL версии 5.5+ (с поддержкой utf8mb4)"),
			array('n',"Добавлен Justified Gallery"),
			array('n',"Добавлен popper.js"),
			array('u','jQuery обновлен до версии 3.3.1'),
			array('u','Bootstrap обновлен до версии 4.0.0'),
			array('u','pjax обновлен до версии 2.0.1'),
			array('u','jScroll обновлен до версии 2.3.9'),
			array('u','Bootstrap Select обновлен до версии 1.13.0-beta'),
		),
		'0.7.5' => array(
			array('u','Bootstrap-select обновлен до версии 1.12.4'),
			array('u','FontAwesome обновлен до версии 5.0.8'),
			array('i','Очередное обновление API Вконтакта.'),
		),
		'0.7.4' => array(
			array('bf','Исправлена ошибка в разделе Сообщения. Если в базе уже имеются данные, следуйте <a href="update/index.html" target="_blank">инструкции</a> на странице обновления.'),
		),
		'0.7.3' => array(
			array('bf',"Частично исправлен fancybox в разделе сообщений. Для некоторых типов сообщений могут остаться ошибки."),
			array('u',"Убраны абсолютные пути из радиректов."),
		),
		'0.7.2' => array(
			array('n',"Видео теперь можно минимизировать и при переходе в другой раздел оно не закроется."),
			array('n',"Добавлен быстрый поиск для видео."),
			array('n',"Настройки видео плеера теперь сохраняются в куки."),
			array('n',"Добавлен плагин js-cookie."),
			array('u',"jQuery обновлен до версии 1.12.4"),
			array('u',"Fancybox обновлен до версии 3.1.20"),
		),
		'0.7.1' => array(
			array('i',"При ошибках скачивания видео, обновите youtube-dl до версии от 2017.05.29 или выше."),
			array('u',"jScroll обновлен до версии 2.3.9"),
			array('u',"Bootstrap обновлен до версии 3.3.7"),
			array('i',"Запросы изменены на MySQLi."),
			array('bf',"Микрофикс ссылок аватаров групп и пользователей в очереди закачки."),
			array('b',"Проблема с отображением записи &laquo;пользователь обновил фото&raquo;."),
		),
		'0.7.0' => array(
			array('n',"Добавлено автоматическое проигрывание gif'ов на стене."),
			array('n',"Добавлена синхронизация документов на стене."),
			array('n',"Добавлена библиотека pjax."),
			array('u',"Bootstrap Select обновлен до версии 1.12.1."),
			array('bf',"Исправлен баг автоматической закачки для аттачей."),
			array('bf',"Сообщение со стены не сохранялось если начиналось с эмодзи."),
		),
	),
	'0.6.2' => array(
		'2017-03-04' => array(
			array('u',"Небольшие изменения в дизайне."),
			array('bf',"На стене могли неверно отображаться фото, видео, музыка групп и пользователей если их id совпадал с id материалами владельца."),
		),
	),
	'0.6.1' => array(
		'2017-01-24' => array(
			array('u',"Добавлена быстрая синхронизация для стены."),
		),
		'2017-01-23' => array(
			array('u',"Добавлены фильтры по дате и длительности для видео."),
			array('u',"Добавлена опция для автовоспроизведения локального видео."),
			array('u',"Переработан дизайн локального видео плеера."),
			array('b',"Полноэкранный режим не поддерживается в IE10 при проигрывании видео."),
		),
		'2017-01-15' => array(
			array('n',"Добавлена возможность скачивать видео из ВКонтакте с авторизацией через youtube-dl."),
		),
	),
	'0.6' => array(
		'2016-12-30' => array(
			array('n',"Добавлена синхронизация документов."),
			array('n',"Добавлена проверка версии структуры базы данных и инструкция по обновлению."),
		),
	),
	'0.5.6' => array(
		'2016-12-28' => array(
			array('d',"Добавлена проверка на отключенный API Вконтакте для музыки при синхронизации стены."),
		),
		'2016-12-19' => array(
			array('d',"API Вконтакте для музыки был отключен. Надеюсь, что через некоторое время он вновь будет доступен для всех."),
		),
	),
	'0.5.5' => array(
		'2016-11-21' => array(
			array('bf',"Поправлены пути для установки скрипта в суб-директорию. Спасибо Ивану за багрепорт."),
		),
		'2016-10-30' => array(
			array('n',"Добавлена быстрая синхронизация (только системные альбомы)."),
		),
		'2016-06-16' => array(
			array('u',"В плейлисте изменена иконка для активного трека."),
		),
		'2016-06-15' => array(
			array('bf',"Исправлен баг с упорядочиванием фото на последней странице."),
			array('bf','Исправлен баг с отображением кнопки "показать все альбомы" при ресайзе окна.'),
		),
		'2016-06-05' => array(
			array('u','Изменен урл загрузки роиков для ВКонтакте, чтобы можно было сохранять "приватные" видео.'),
			array('bf',"Исправлен баг в поиске видео когда при прокрутке следующая страница загружалась без фильтра."),
		),
	),
	'0.5' => array(
		'2016-06-01' => array(
			array('n',"Добавлена возможность поиска видео по типу, сервису и качеству."),
		),
		'2016-05-29' => array(
			array('u',"Добавлена настройка для проигрывания локального видео по умолчанию."),
			array('n',"Добавлен плеер для локальных видео."),
			array('bf',"Исправлено сохранение локального пути для видеофайла."),
		),
		'2016-05-28' => array(
			array('u',"Раздел видео обновлен."),
			array('n','Добавлена поддержка создания локальной копии видеофайлов с сервисов YouTube и VK.com при помощи <a href="https://github.com/rg3/youtube-dl" target="_blank"><i class="fa fa-link"></i> youtube-dl</a> (необходимо установить отдально!)'),
		),
		'2016-05-24' => array(
			array('u',"Улучшена сортировка плейлиста по исполнителю и названию трека."),
		),
		'2016-05-19' => array(
			array('n',"Добавлены красивые списки (Bootstrap Select)"),
			array('n',"Плеер audio.js заменен на более продвинутый jPlayer.js в связи с этим добавились новые опции а так же изменился дизайн"),
			array('u',"В аудиозаписях добавлено отображение удаленных треков"),
			array('n',"Добавлена синхронизация альбомов для аудиозаписей."),
		),
	),
	'0.4.5' => array(
		'2016-05-12' => array(
			array('n',"Добавлен perfect-scrollbar.js."),
			array('n',"Добавлен вывод альбомов (в которых есть хотя бы одно изображение) с превьюшками."),
			array('u',"В альбомах фото выводятся в порядке убывания."),
		),
		'2016-05-07' => array(
			array('u',"Для синхронизации аттачей добавлена проверка на наличие файла локально."),
			array('bf',"Исправлено сохранение ссылки для репоста."),
			array('bf',"Исправлено сохранение аудиозаписи для репоста."),
		),
	),
	'0.4' => array(
		'2016-05-06' => array(
			array('n',"Добавлена возможность смотреть отдельные посты со стены в лайтбоксе."),
			array('n',"Хэш навигация (hashnav.js)."),
			array('bf',"Исправлен баг с зависанием загрузки стены в хроме."),
		),
		'2016-05-05' => array(
			array('bf',"Добавлена дополнительная проверка на контент для закачек."),
			array('u',"Добавлен код ошибки если закачка не удалась."),
			array('u',"Добавлен интерактивный отсчет для таймеров."),
			array('bf',"Исправлено отображение чекбокса в настройках для IE 11."),
			array('n',"Синхронизация сообщений (стена): аудиозаписи."),
			array('n',"Синхронизация сообщений (стена): репосты (в том числе вложенные)."),
			array('n',"Синхронизация сообщений (стена): ссылки."),
		),
		'2016-05-04' => array(
			array('u',"Множественные изменения в коде."),
			array('u',"Опция для виртуальных хостов вынесена отдельно. Теперь для её включения необходимо отредактировать конфиг и одну функцию вместо множества файлов."),
			array('n',"Синхронизация сообщений (стена): добавлен парсинг видео аттачей, а так же обработка репостов (фото, видео)."),
		),
		'2016-04-30' => array(
			array('u',"Общий стиль изменен под новый дизайн ВК."),
			array('bf',"Не работал оптимизатор изображений в альбомах на FireFox."),
		),
		'2016-04-29' => array(
			array('n',"Добавлен раздел «Сообщения»."),
			array('n',"Синхронизация сообщений (стена). Альфа-версия."),
		),
		'2016-04-28' => array(
			array('n',"Добавлены настройки (Авто-редиректы)"),
			array('n',"Красивые чекбоксы (Bootstrap Toggle)"),
		),
	),
	'0.3' => array(
		'2016-04-24' => array(
			array('u',"Добавлена полоса прогресса при синхронизации фотографий."),
		),
		'2016-04-21' => array(
			array('u',"Обновление названий альбомов при синхронизации."),
		),
		'2016-03-27' => array(
			array('bf',"Исправлен баг при синхронизации если альбом пустой."),
		),
		'2016-03-21' => array(
			array('u',"Код немного приведен в порядок для работы на PHP 5.3.x."),
			array('bf',"Неправильные флаги при повторной синхронизации музыки."),
		),
		'2016-03-19' => array(
			array('u',"Настройки таймаутов перенесены в конфиг."),
			array('u',"Обновление количества фото в альбомах при синхронизации."),
			array('u',"Не отображалась кнопка автоматического скачивания."),
		),
		'2016-03-15' => array(
			array('n',"Просмотр видеозаписей"),
			array('u',"При ошибке синхронизации в автоматическом режиме и html ответе сервера (прим. ошибка Bad Gateway) добавлен пропуск данных ID."),
			array('n',"Добавлена синхронизация видеозаписей"),
			array('bf',"Исправлено отображение смешенного контента в очереди закачек."),
		),
	),
	'0.2' => array(
		'2016-03-14' => array(
			array('n',"Добавлен плеер в раздел музыки (audio.js)"),
			array('bf',"Добавлена проверка на содержимое ответа сервера."),
			array('bf',"Некоторые кириллические названия треков не сохранялись"),
		),
		'2016-03-13' => array(
			array('n',"Добавлены иконки (FontAwesome)"),
			array('i',"Имена аудиозаписей чистятся от не ASCII символов, так как PHP для Windows более чем точно будет скомпилирован без поддержки юникода."),
			array('b',"Непонятно почему ВК не отдает некоторые аудиозаписи через API, хотя они не являются заблокированными."),
			array('u',"Обновлен блок и страница очереди закачки. Объединены фотографии и аудиозаписи."),
			array('n',"Синхронизация аудиозаписей"),
		),
		'2016-03-12' => array(
			array('u',"Раздел Альбомы по-умолчанию отображают 25 последних добавленных фотографий"),
			array('bf',"Добавлена проверка на размер файла при сохранении"),
		),
	),
	'0.1' => array(
		'2016-03-09' => array(
			array('n',"Просмотр фотографий (Fancyfox)"),
			array('b',"Необходима проверка на размер файла при закачке"),
			array('u',"Ссылку на очередь отображается если есть хотя бы один элемент"),
			array('n',"Подгрузка фотографий при скроллинге (jscroll.js)"),
			array('n',"Responsive макет для фотографий (freewall.js)"),
			array('n',"Отображение фотографий в альбомах"),
		),
		'2016-03-08' => array(
			array('n',"Получение и сохранение фото"),
			array('n',"Очередь закачки"),
			array('u',"Блок очереди закачки в панели управления"),
		),
		'2016-03-07' => array(
			array('fx',"Синхронизация фотографий"),
			array('bf',"Исправлены баги в синхронизации альбомов"),
		),
		'2016-03-06' => array(
			array('fx',"Синхронизация альбомов"),
			array('fx',"Авторизация"),
			array('n',"Интерфейс, структура"),
			array('i',"Начало разработки"),
		),
	),
);

// twig
$skin->twig_set('libs',$libs);
$skin->twig_set('changelog',$changelog);
echo $skin->twig_render('about');
// twig end

if(!$cfg['pj']){
	print $skin->footer(array('extend'=>''));
}

$db->close($res);

?>